contrastscan-remote/src/main/java/com/contrastsecurity/scan/ContrastRemoteScanner.kt

```diff
@@ -0,0 +1,193 @@
+    val sdk: ContrastSDK = connectToContrast(creds, apiUrl)
+
+    fun scan(prescan: Path, artifact: Path, scanOutputPath: Path, timeoutMs: Long) {
+        val project: Project = findOrCreateProject()
+
+        logger.info("Uploading " + artifact.fileName + " to Contrast Scan")
+
+        val codeArtifact: CodeArtifact
+        codeArtifact = try {
+            project.codeArtifacts().upload(artifact, prescan)
+        } catch (e: IOException) {
+            throw RuntimeException("Failed to upload code artifact to Contrast Scan", e)
+        } catch (e: HttpResponseException) {
+            throw RuntimeException("Failed to upload code artifact to Contrast Scan", e)
+        }
+
+        logger.info("Starting scan for project [${projectName}] with label [$labelName]")
+        val scan: Scan
+        scan = try {
+            project.scans().define().withLabel(labelName).withExistingCodeArtifact(codeArtifact).create()
+        } catch (e: IOException) {
+            throw RuntimeException("Failed to start scan for code artifact $codeArtifact", e)
+        } catch (e: HttpResponseException) {
+            throw RuntimeException("Failed to start scan for code artifact $codeArtifact", e)
+        }
+
+        // show link in build log
+        val clickableScanURL: URL = createClickableScanURL(project.id(), scan.id())
+        logger.info("Scan results will be available at " + clickableScanURL.toExternalForm())
+
+        logger.info("Waiting for scan results...")
+        waitForResults(scan, scanOutputPath, timeoutMs)
+        //logSummary(summary)
+    }
+
+    private fun findOrCreateProject(): Project {
+        val projects = sdk.scan(creds.orgId).projects()
+        val maybeProject = projects.findByName(projectName)
+        val project: Project = if (maybeProject.isEmpty) {
+            logger.info("Creating project with name: '$projectName'")
+            projects.define().withName(projectName).withLanguage("JAVA").create()
+        } else {
+            maybeProject.get()
+        }
+
+        if (project.archived()) {
+            // TODO the behavior of tools like this plugin has yet to be defined with respect to
+            //  archived projects; however, the UI exposes no way to archive projects at the moment.
+            //  For now, simply log a warning to help debug this in the future should we encounter this
+            //  case
+            throw IllegalArgumentException("Project $projectName is archived")
+        }
+        return project
+    }
+
+    private fun createClickableScanURL(projectId: String, scanId: String): URL {
+        val path = java.lang.String.join(
+            "/",
+            "",
+            "Contrast",
+            "static",
+            "ng",
+            "index.html#",
+            creds.orgId,
+            "scans",
+            projectId,
+            "scans",
+            scanId
+        )
```

Comment:
Why this roundabout way of concatenating the string? `val path = "/Contrast/static/ng/index.html#/${creds.orgId}/scans/${projectId}/scans/${scanId}`?